### ISSUE TITLE
[PRE REVIEW]: Augur: a bioinformatics toolkit for phylogenetic analyses of human pathogens

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2901]
**Submitting author:** @huddlej (<a href="http://orcid.org/0000-0002-4250-2063">John Huddleston</a>)
**Repository:** <a href="https://github.com/nextstrain/augur/" target ="_blank">https://github.com/nextstrain/augur/</a>
**Version:** v10.1.1
**Editor:** @majensen
**Reviewers:** @dcnickle, @Maghnuso
**Managing EiC:** Kevin M. Moerman

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**Author instructions**

Thanks for submitting your paper to JOSS @huddlej. **Currently, there isn't an JOSS editor assigned** to your paper.

@huddlej if you have any suggestions for potential reviewers then please mention them here in this thread (without tagging them with an @). In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission (please start at the bottom of the list).

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Failed to discover a `Statement of need` section in paper

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.20944/preprints202008.0220.v1 is OK
- 10.2807/1560-7917.ES.2020.25.32.2001410 is OK
- 10.1186/s13059-015-0726-x is OK
- 10.1371/journal.pone.0009490 is OK
- 10.1093/bioinformatics/btu033 is OK
- 10.1038/s41591-020-0935-z is OK
- 10.1126/science.abc0523 is OK
- 10.1093/bioinformatics/bty407 is OK
- 10.1093/bioinformatics/btp352 is OK
- 10.1093/bioinformatics/btr509 is OK
- 10.1093/nar/gkf436 is OK
- 10.1093/molbev/msu300 is OK
- 10.1093/ve/vex042 is OK

MISSING DOIs

- 10.1093/bioinformatics/btv381 may be a valid DOI for title: nextflu: real-time tracking of seasonal influenza virus evolution in humans.
- 10.1093/bioinformatics/bts480 may be a valid DOI for title: Snakemake—a scalable bioinformatics workflow engine

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.82 s (248.2 files/s, 46903.2 lines/s)
--------------------------------------------------------------------------------
Language                      files          blank        comment           code
--------------------------------------------------------------------------------
JSON                             35              0              0          17466
Python                           70           2451           3466           9102
Markdown                         26           1004              0           2820
TeX                               2             33             19            732
reStructuredText                 57            200            423            303
Bourne Again Shell                4             52             35            200
YAML                              3              5              4             59
Bourne Shell                      2             14              5             41
DOS Batch                         1              8              1             26
INI                               1              8              0             22
make                              2              7              9             14
CSS                               1              3              0             12
--------------------------------------------------------------------------------
SUM:                            204           3785           3962          30797
--------------------------------------------------------------------------------


Statistical information for the repository 'ce1ec198c17eccffe3b9555f' was
gathered on 2020/12/11.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Akshay Subramanian               1            59              0            0.07
Alli Black                       4            18             14            0.04
Barney Potter                   28           292            147            0.49
Cameron Devine                   2             8              2            0.01
Chris Woszczak                   1             1              1            0.00
Christian Clauss                 2             3              3            0.01
Christopher Tomkins-             1             8              0            0.01
Danielle Kain                    1            55              2            0.06
Eddie Lebow                     20          1532            444            2.22
Emma Hodcroft                  121          2343            762            3.49
Eric Danielson                  82          1667            458            2.39
Gytis Dudas                      4            97             66            0.18
James Hadfield                 270         10395          15027           28.58
John Huddleston                381          8133           2965           12.48
Jover                           14          1073            796            2.10
Julien BORDELLIER                2             6             11            0.02
Kairsten Fay                     4           558            389            1.06
Louise Moncla                    4            39             23            0.07
Mathias Walter                   1            17             21            0.04
Mingye Wang                      5            40             27            0.08
PierreBarrat                     1             8              4            0.01
Rhys Kidd                        1             1              1            0.00
Richard Neher                  445         15997           5863           24.58
Ryan Grout                      14           121             49            0.19
Samuel Zhang                     2            64              2            0.07
Sidney Bell                    112          3000           2591            6.29
Thomas A Caswell                 2            32             19            0.06
Thomas Sibley                   79          1100           3577            5.26
Tony Tung                        1            92              3            0.11
Trevor Bedford                 381          4819           3801            9.69
Venkata Sai Kiran Ko             8           125             70            0.22
eharkins                         3            21              3            0.03
root                             1            44             14            0.07
terrycojones                     3            13             13            0.03

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Barney Potter                38           13.0         18.2                2.63
Cameron Devine                7           87.5          8.4                0.00
Chris Woszczak                1          100.0          6.8                0.00
Christian Clauss              3          100.0          7.7                0.00
Christopher Tomkins-          4           50.0          4.3                0.00
Danielle Kain                55          100.0          7.4               10.91
Eddie Lebow                1465           95.6          5.5                5.19
Elias Harkins                20          100.0          0.7                5.00
Emma Hodcroft               904           38.6         25.0                9.18
Eric Danielson             1402           84.1          7.1                7.77
Gytis Dudas                  50           51.5         29.5                6.00
James Hadfield             2434           23.4         21.2                8.30
John Huddleston            2542           31.3         21.8               14.24
Jover                       506           47.2         16.3                6.32
Julien BORDELLIER             6          100.0          6.8                0.00
Kairsten Fay                  6            1.1         16.2                0.00
Mathias Walter                5           29.4          7.8               40.00
Mingye Wang                   7           17.5          9.6                0.00
PierreBarrat                  7           87.5         18.3                0.00
Rhys Kidd                     1          100.0          7.5              100.00
Richard Neher              3731           23.3         24.8                8.15
Ryan Grout                  105           86.8          7.6                0.95
Samuel Zhang                 62           96.9          7.1               14.52
Sidney Bell                  98            3.3         39.8                2.04
Thomas A Caswell             31           96.9          7.7                9.68
Thomas Sibley               616           56.0         22.8               17.37
Tony Tung                    92          100.0          3.7                7.61
Trevor Bedford              608           12.6         22.8               14.47
Venkata Sai Kiran Ko         32           25.6          0.3                9.38
root                         39           88.6          7.6                5.13
terrycojones                  3           23.1          7.6                0.00
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02901/joss.02901/10.21105.joss.02901.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02901/joss.02901/10.21105.joss.02901.pdf) :page_facing_up: :point_left:

**Comment by Kevin-Mattheus-Moerman** [comment]
@majensen @csoneson @will-rowe @jni could one of you handle this submission? Thanks

**Comment by Kevin-Mattheus-Moerman** [comment]
@huddlej thanks for your submission. Our system is detecting you paper lacks a `Statement of need` section which is a requirement (https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain). Can you work to add it? Once you've done that you can call the following in a comment here to update the paper: 
`@whedon generate pdf`

**Comment by jni** [comment]
@Kevin-Mattheus-Moerman sorry, I'm going to be on leave for about a month starting this week — just working frantically to wrap up my ongoing papers-in-review. I'll reboot again mid-January.

**Comment by majensen** [comment]
@whedon set me as editor

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by majensen** [comment]
@whedon assign me as editor

**Comment by whedon** [comment]
OK, the editor is @majensen

**Comment by majensen** [comment]
@Maghnuso are you up for another JOSS review?

**Comment by majensen** [comment]
@rvosa any chance you would be able to review this software for JOSS? Appreciate your consideration!

**Comment by majensen** [comment]
@whedon assign @dcnickle as reviewer

**Comment by whedon** [comment]
OK, @dcnickle is now a reviewer

**Comment by Maghnuso** [comment]
> @Maghnuso are you up for another JOSS review?

Hi Mark, yes, I'd be happy to. It's not as if I don't need the practice :-) 
I won't be able to touch it for a few days though, if that's ok... 
-M

**Comment by majensen** [comment]
That's the spirit @Maghnuso ! Much appreciated

**Comment by majensen** [comment]
@whedon add @Maghnuso as reviewer

**Comment by whedon** [comment]
OK, @Maghnuso is now a reviewer

**Comment by majensen** [comment]
@huddlej @dcnickle @Maghnuso we have people in place in record time, and I will start the review. Please contact me with @majensen in the review or directly via `maj -at- fortinbras -dot- us` if you have any questions during the process. Thanks!

**Comment by majensen** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/2906.

**Comment by majensen** [comment]
See everyone in #2906 - thanks again.

**Comment by DavidNickle** [comment]
I've started my review of #2901 - I appreciate the opportunity!

**Comment by DavidNickle** [comment]
**Conflict of interest:** I confirm that I have read the JOSS conflict of interest policy and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.
Code of Conduct

**Repository:** The source code for this software is available at the repository url

**License:** The repository containa a plain-text LICENSE file with the contents of an OSI approved software license.

**Contribution and authorship:** The submitting author made major contributions to the software and the full list of paper authors seem appropriate and complete.

**Functionality:** The software is functional and important adding very important knowledge to the public domain

**Installation:** The installation procedure follows the documentation clearly.  I performed all the task in the Zika_Tuotrial

**Functionality:** I confirmed the functional claims of the on the tutorial data as well as some of my own.

**Performance:** No claims here but it is pretty quick!

**A statement of need:** The authors have a clear audience in mine and the paper and the software should be readably available to that audience.

**Installation instructions:** I ran into an error that clearly was my own (may conda app and phython3 where out of date)! Updated and everything worked including all of the dependencies which are all clearly stated.

**Example usage:** I went through the zika_tutorial and all the functionality worked.  I found this to be fantastic exercise and I will be using Nexstrain in my workflows going forward.

**Automated tests:** YES — tests are available 
Community guidelines: Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support. **YES**

**Software paper:** **YES**

Summary: Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided? **YES**

**A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is? **YES**

**State of the field:** Do the authors describe how this software compares to other commonly-used packages? YES
Quality of writing: Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)? YES
References: Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper citation syntax? **YES**

Lastly, I would like to state that is a piece of scientific coding that is very useful especially to field of phylogenetic and epidemiology. It drives towards the mantra of repeatable results with code that is highly documented and user friendly to the data scientist. The author's have done an exceptional work at sticking to this theme.
